### PR TITLE
fix for existing users logging from frontend

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue # .
https://github.com/joomla/joomla-cms/pull/20051#issuecomment-380207363

### Testing Instructions
for existing users
when logging with an existing user you are  redirected to the edit profile 
 when you click on I agree on the privacy field
now  the profile submission is saved

### CODE REVIEW

**Comment by sandewt** [comment]
**I have tried this**: see comment #20051.

The trailing is not correct.

Sorry this is not the correct way to show the code.

```
public function onContentPrepareForm($form, $data)
	{
		if (!($form instanceof JForm))
		{
			$this->_subject->setError('JERROR_NOT_A_FORM');

			return false;
		}

		// Check we are manipulating a valid form - we do not display this in the admin users form or profile view.
		$name   = $form->getName();
		$layout = $this->app->input->get('layout', 'default');
		$view   = $this->app->input->get('view', 'default');

		if (!in_array($name, array('com_admin.profile', 'com_users.profile', 'com_users.registration')) || $layout != "edit" && $view != "registration")
		{
			return true;
		}

		// Add the registration fields to the form.
		JForm::addFormPath(__DIR__ . '/privacyconsent');
		$form->loadFile('privacyconsent');

		$fields = array(
			'privacy',
		);

		if (is_object($data))
		{
			$userId = isset($data->id) ? $data->id : 0;

			if ($userId > 0)
			{
				// Load the profile data from the database.
				$db = JFactory::getDbo();

				$query = $db->getQuery(true)
					->select($db->quoteName('profile_value'))
					->from($db->quoteName('#__user_profiles'))
					->where($db->quoteName('user_id') . ' = ' . (int) $userId)
					->where($db->quoteName('profile_key') . ' LIKE ' . $db->quote('consent'))
					->order($db->quoteName('ordering') . ' ASC');
				$db->setQuery($query);

				try
				{
					$results = $db->loadRowList();
				}
				catch (RuntimeException $e)
				{
					$this->_subject->setError($e->getMessage());

					return false;
				}

				if (!empty($results[0]) == 1)
				{
					$form->removeField('privacy', 'privacyconsent');

					return true;
				}
			}
		}

		$privacyarticle = $this->params->get('privacy_article');
		$privacynote    = $this->params->get('privacy_note');

		// Push the privacy article ID into the privacy field.
		$form->setFieldAttribute('privacy', 'article', $privacyarticle, 'privacyconsent');
		$form->setFieldAttribute('privacy', 'note', $privacynote, 'privacyconsent');
	}

```

**Comment by alikon** [comment]
@sandewt doesn't matter
added to this pr
thanks

File: plugins/user/privacyconsent/privacyconsent.php [hunk 1]
```diff
@@ -85,6 +85,44 @@ public function onContentPrepareForm($form, $data)
 			'privacy',
 		);
 
+		if (is_object($data))
+		{
+
```
Comment (Quy) [inline plugins/user/privacyconsent/privacyconsent.php]: Remove blank line

**Comment by sandewt** [comment]
Thanks @alikon and @Quy .

Line 117:

```
if (!empty($results[0]) == 1)
```
Should be (better)
```
if (!empty($results[0]))

```

File: plugins/user/privacyconsent/privacyconsent.php (hunk 1 above)
Comment (alikon) [reply plugins/user/privacyconsent/privacyconsent.php]: done

**Comment by alikon** [comment]
@sandewt changed the query a little bit

**Comment by sandewt** [comment]
@alikon / @brianteeman

At first this seemed to be a solution, but the problem is only being moved. Sad.

But when you (re) save (Edit Your Profile) the $form variable is emptied, after which the known warning message appears.

**Comment by sandewt** [comment]
>better check

Thanks @alikon ,  works fine.

Rule 148: `$option` is not used anymore. 

```
$option = $this->app->input->getCmd('option');
```

(The tooltip does not display the text correctly. )

**Comment by brianteeman** [comment]
I have merged this code. my feeling was that it is not correct to only show the privacyconsent field on the edit profile field when the user has previously consented and that they should consent each time they update their information. However I am not that bothered about that so I have left it with your code

THANKS!!!

**Comment by alikon** [comment]
thanks for your patience

we can always make a new pr after the things become running in the wild

**Pull request merged**